kore/src/Kore/Strategies/Verification.hs

```diff
@@ -271,20 +306,24 @@ verifyClaim
 
     throwUnproven
         :: LogicT (Verifier simplifier) CommonProofState
-        -> Verifier simplifier ()
+        -> Verifier simplifier [CommonProofState]
```

Comment:
This list could possibly be very large, so we don't want to collect it in memory. `throwUnprovenOrElse` should return the largest depth, which we can log here. `throwUnproven` can return `()` as before.